### ISSUE TITLE
[Handshake] Allow different transport protocols 

### ISSUE THREAD

**Issue by tiero** [TDex-network/tdex-specs#16]
RE: BOTD#2

TL;DR (bu read anyhow) we need to support multiple transport, not a single one.

Long explanation:
for browsers and mobile app to consume the daemon rpcs would be to necessary to have also server-side TLS, otherwise would be marked as insecure by browser and not allowed in app stores by mobile apps. So this means that if we support Noise-like protocols, we should mandatory support also TLS. I guess this is doable, but this led to the question if we should impose a strict secure transport over another. Noise + TLS would also make it slower (ie double encryption), without considering instead that a lot of peers would want to use tor hidden services too.

My proposal:

- rename Handshake protocol to Transport protocol
- each peer should advertise (via cleartext HTTP1 endpoint) which set of protocols supports
- It should be a one (or couple) between: insecure, onion, TLS, mTLS, Noise
